File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.py
import sys

input = sys.stdin.readline
n, k = map(int, input().split())
circle = [x for x in range(1, n + 1)]
removed = []
now = k - 1
while circle:
    removed.append(circle.pop(now))
    if circle:
        now = ((now - 1) + k) % len(circle)
print(f'<{", ".join(map(str,removed))}>')

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.js
const path = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = require('fs').readFileSync(path).toString().trim().split('\n');

function solution(input) {
  const [N, K] = input[0].split(' ').map(Number);
  const answer = [];
  const queue = Array.from({ length: N }, (_, i) => i + 1);
  let index = 0;

  while (queue.length > 0) {
    index = (index + K - 1) % queue.length; // K번째 사람 찾기 (원형 큐)
    answer.push(queue.splice(index, 1)[0]); // 해당 인덱스의 사람 제거
  }

  console.log(`<${answer.join(', ')}>`);
}

solution(input);